app/src/main/java/com/android/voyageur/ui/trip/activities/ActivitiesMapScreen.kt

```diff
@@ -0,0 +1,275 @@
+        tripsViewModel.getActivitiesForSelectedTrip().filter { activity ->
+          val matchesQuery =
+              activity.title.contains(searchQuery, ignoreCase = true) ||
+                  activity.description.contains(searchQuery, ignoreCase = true)
+          val matchesFilters =
+              selectedFilters.isEmpty() ||
+                  selectedFilters.contains(activity.activityType.toString())
+          val matchesDate =
+              selectedDate == null ||
+                  activity.startTime.toDate().let { date ->
+                    date.year == selectedDate!!.year &&
+                        date.month == selectedDate!!.month &&
+                        date.date == selectedDate!!.date
+                  }
+          matchesQuery && matchesFilters && matchesDate
+        }
+    onActivitiesChanged(activities)
+  }
+
+  Column {
+    // Search Bar with Filter Button
+    Row(
+        modifier = Modifier.fillMaxWidth().padding(16.dp),
+        verticalAlignment = Alignment.CenterVertically) {
+          SearchBar(
+              placeholderId = R.string.search_activities,
+              onQueryChange = { query ->
+                searchQuery = query
+                applyFilters()
+              },
+              modifier = Modifier.weight(1f).testTag("searchBar"))
+          Spacer(modifier = Modifier.width(8.dp))
+          IconButton(
+              modifier = Modifier.testTag("filterButton"), onClick = { showFilterDialog = true }) {
+                Icon(
+                    imageVector = Icons.Outlined.FilterAlt,
+                    contentDescription = stringResource(R.string.filter_activities),
+                    tint = MaterialTheme.colorScheme.primary)
+              }
+        }
+
+    // Google Map
+    Box(
+        modifier =
+            Modifier.fillMaxSize()
+                .padding(16.dp)
+                .clip(MaterialTheme.shapes.medium)
+                .background(MaterialTheme.colorScheme.secondary)) {
+          GoogleMap(
+              modifier = Modifier.fillMaxSize().testTag("GoogleMap"),
+              cameraPositionState = cameraPositionState) {
+                activities.forEach { activity ->
+                  Marker(
+                      state =
+                          MarkerState(
+                              position = LatLng(activity.location.lat, activity.location.lng)),
+                      title = activity.title,
+                      snippet = activity.description)
+                }
+              }
+        }
+  }
+
+  // Filter Dialog
+  if (showFilterDialog) {
+    FilterDialog(
+        onFilterByDay = { date ->
+          selectedDate = date
+          applyFilters()
+          showFilterDialog = false
+        },
+        onFilterByType = { filters ->
+          selectedFilters = filters
+          applyFilters()
+          showFilterDialog = false
+        },
+        onClearFilters = {
+          activities = tripsViewModel.getActivitiesForSelectedTrip()
```

Comment:
When we clear the filters, I think that also the text in the search box should be cleared, as it's more intuitive.

Reply:
The search bar is a separate component with a separate clear button. Actually I believe that it is more intuitive that each button only affects its component. But we could further discuss it with the team.

Reply:
For now is ok, we'll discuss it with the team and maybe address it in the UI fixes PR.